scripts/db_migrator.py

```diff
@@ -0,0 +1,136 @@
+
+import sys
+import argparse
+import syslog
+from swsssdk import ConfigDBConnector
+
+
+SYSLOG_IDENTIFIER = 'db_migrator'
+
+
+def log_info(msg):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+
+def log_error(msg):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+
+class DBMigrator():
+    def __init__(self):
+        """
+        Version string format:
+           version_<major>_<minor>_<build>
+              major: starting from 1, sequentially incrementing in master
+                     branch.
+              minor: in github branches, minor version stays in 0. This minor
+                     version creates space for private branches derived from
+                     github public branches. These private branches shall use
+                     none-zero values.
+              build: sequentially increase within a minor version domain.
+        """
+        self.CURRENT_VERSION = 'version_1_0_1'
+
+        self.TABLE_NAME      = 'VERSIONS'
+        self.TABLE_KEY       = 'DATABASE'
+        self.TABLE_FIELD     = 'VERSION'
+        self.configDB        = ConfigDBConnector()
+        self.configDB.db_connect('CONFIG_DB')
+
+
+    def migrate_pfc_wd_table(self):
+        # Migrate all data entries from table PFC_WD_TABLE to PFC_WD
+        data = self.configDB.get_talbe('PFC_WD_TABLE')
+        for key in data.keys():
+            self.configDB.set_entry('PFC_WD', key, data[key])
+        self.configDB.delete_table('PFC_WD_TABLE')
+
+
+    def version_unknown(self):
+        """
+        version_unknown tracks all SONiC versions that doesn't have a version
+        string defined in config_DB.
+        Nothing can be assumped when migrating from this version to the next
+        version.
+        Any migration operation needs to test if the DB is in expected format
+        before migrating date to the next version.
+        """
+
+        log_info('Handling version_unknown')
+
+        # NOTE: Uncomment next 3 lines of code when the migration code is in
+        #       place. Note that returning specific string is intentional,
+        #       here we only intended to migrade to DB version 1.0.1.
+        #       If new DB version is added in the future, the incremental
+        #       upgrade will take care of the subsequent migrations.
+        # self.migrate_pfc_wd_table()
+        # self.set_version('version_1_0_1')
+        # return 'version_1_0_1'
+
+
+    def version_1_0_1(self):
+        """
+        Current latest version. Nothing to do here.
+        """
+        log_info('Handling version_1_0_1')
+
+        return None
+
+
+    def get_version(self):
+        version = self.configDB.get_entry(self.TABLE_NAME, self.TABLE_KEY)
+        if version and version[self.TABLE_FIELD]:
+            return version[self.TABLE_FIELD]
+
+        return 'version_unknown'
+
+
+    def set_version(self, version=CURRENT_VERSION):
```

Comment:
Cannot use self.CURRENT_VERSION here. Define as None and set to self.CURRENT_VERSION inside function when None.